lib/full-site-editing/block-templates.php

```diff
@@ -269,8 +352,9 @@ function gutenberg_get_block_templates( $query = array(), $template_type = 'wp_t
 				true
 			);
 			$should_include = false === $is_custom && (
-				! isset( $query['slug__in'] ) || in_array( $template_file['slug'], $query['slug__in'], true )
-			);
+				! isset( $query['slug__in'] ) || in_array( $template_file['slug'], $query['slug__in'], true ) ) && (
+					! isset( $query['section'] ) || $template_file['section'] === $query['section']
+				);
```

Comment:
Duh, this is difficult to read. Maybe we should split this into a few variables?

Reply:
Updated, should be more clear now.